pkgs/development/tools/volta/default.nix

```
@@ -0,0 +1,38 @@
+{
+  lib,
+  rustPlatform,
+  libiconv,
+  stdenv,
+  darwin,
+  fetchFromGitHub,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "volta";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "volta-cli";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-+j3WRpunV+3YfZnyuKA/CsiKr+gOaP2NbmnyoGMN+Mg=";
+  };
+
+  cargoHash = "sha256-g6Ai0QN9cDiNX/YHdMM7/llJ/pCWZtv66wTaKvuN+bE=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [darwin.apple_sdk.frameworks.Security libiconv];
+
```

In case you need any help with the completions, I had been working on a derivation for this as well:

```suggestion
  buildInputs = [installShellFiles]
  ++ lib.optionals stdenv.isDarwin [darwin.apple_sdk.frameworks.Security libiconv];

  postInstall = ''
    installShellCompletion --cmd volta \
      --bash <($out/bin/volta completions bash) \
      --fish <($out/bin/volta completions fish) \
      --zsh <($out/bin/volta completions zsh)
  '';
```